test/solidity-ibc/IBCAdminTest.t.sol

```diff
@@ -142,4 +151,52 @@ contract IBCUUPSUpgradeableTest is Test {
         vm.expectRevert(abi.encodeWithSelector(IIBCUUPSUpgradeableErrors.Unauthorized.selector));
         ics26Router.setTimelockedAdmin(newTimelockedAdmin);
     }
+
+    function test_success_pauseAndUnpause() public {
+        assertEq(ics20Transfer.getPauser(), ics20Pauser);
+
+        vm.prank(ics20Pauser);
+        ics20Transfer.pause();
+        assert(ics20Transfer.paused());
+
+        // Try to call a paused function
+        IICS20TransferMsgs.SendTransferMsg memory sendMsg;
+        vm.expectRevert(abi.encodeWithSelector(PausableUpgradeable.EnforcedPause.selector));
+        ics20Transfer.sendTransfer(sendMsg);
+
+        vm.prank(ics20Pauser);
+        ics20Transfer.unpause();
+        assert(!ics20Transfer.paused());
+    }
+
+    function test_failure_pauseAndUnpause() public {
+        vm.expectRevert(abi.encodeWithSelector(IIBCPausableUpgradeableErrors.Unauthorized.selector));
+        ics20Transfer.pause();
+        assert(!ics20Transfer.paused());
+
+        vm.prank(ics20Pauser);
+        ics20Transfer.pause();
+        assert(ics20Transfer.paused());
+
+        vm.expectRevert(abi.encodeWithSelector(IIBCPausableUpgradeableErrors.Unauthorized.selector));
+        ics20Transfer.unpause();
+        assert(ics20Transfer.paused());
+    }
+
+    function test_success_setPauser() public {
+        address newPauser = makeAddr("newPauser");
+
+        ics20Transfer.setPauser(newPauser);
+        assertEq(ics20Transfer.getPauser(), newPauser);
```

Comment:
probably would be good to verify that you cannot call it with the old pauser, and that you _can_ call it with the new one.

Reply:
I think `getPauser()` is enough since there is only a single state entry in the contract for pauser. This is a unit test, and the goal of the function is to change that state entry